src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
```

Comment:
PullProofAI

Consider adding a comment to describe the purpose of the `ResponseInterceptor` class.

```diff
+ // Interceptor for modifying response structure and logging details
```



<!-- This is an auto-generated comment by PullProofAI -->